src/recv.c

```diff
@@ -32,10 +32,41 @@
 #include "internal.h"
 #include "iasync.h"
 #include "sync.h"
+#include "binary.h"
 
 #include <string.h>
 #include <stdlib.h>
 
+char * 
+mpd_recv_binary(struct mpd_connection *connection, const unsigned binary) {
+	char * data = malloc(binary);
+	assert(data);
+        unsigned consumed = 0;
+        
+        while (consumed < binary) {
+                struct mpd_binary src = mpd_sync_recv_binary(connection->async,
+                                                             mpd_connection_timeout(connection), 
+                                                             binary - consumed);
+                assert(src.data != NULL);
```

Comment:
This makes the program crash after an I/O error (debug build) or loop endlessly (release build). Both are the worst imaginable outcomes.